src/System.Memory/src/System/SpanHelpers.byte.cs

```diff
@@ -119,6 +130,39 @@ public static unsafe int IndexOf(ref byte searchSpace, byte value, int length)
             return (int)(byte*)(index + 6);
         Found7:
             return (int)(byte*)(index + 7);
+#if !netstandard10
+        VectorLength:
+            // Already checked, but for jit branch elmination
```

Comment:
The JIT has dead code elimination. When no vector support, the branch at the top should be enough to make this code. I do not think that this extra check can ever help anything.